ci/clang_tidy.sh

```diff
@@ -46,7 +46,12 @@ else
   fix_flag="--fix --lint-all"
 fi
 
-COMPILE_COMMANDS="$SRC_DIR/out/host_debug/compile_commands.json"
+# Determine wether to use x64 or arm64.
+if command -v arch &> /dev/null && [[ $(arch) == "arm64" ]]; then
+  CLANG_TIDY_PATH="buildtools/mac-arm64/clang/bin/clang-tidy"
+fi
+
+COMPILE_COMMANDS="$SRC_DIR/out/$BUILD_DIR/compile_commands.json"
```

Comment:
How is `$BUILD_DIR` defined?

Reply:
It doesn't.  This effectively makes gn get called every time which isn't an error.  In fact it might be desired.  It's quick to execute (1704ms) and means you'll get coverage from newly added files.  I'm going to delete the check.

Reply:
`gn` run time can vary a lot. Some runs on my M1 take >10s. Here's a run on CI taking >4s: https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20Production%20Engine%20Drone/194351/overview.

Sorry to nitpick, but would it be okay to just fix the arm vs. intel issue in this PR?

Reply:
Yea, no problem.  I brought back the check.  What is it doing in all that 10s?  In CI we should have just run `gn`... Oh I guess it won't match though because of the `--force-mac-arm64`.  How I loathe that flag.

Reply:
I actually for a moment considered making that check smarter.  It should be doing a timestamp check against the repository.  That will take a fair bit of work though so I left it for a future exercise.  The indiscriminate `gn` doesn't have that bug.

Reply:
The `clang_tidy` script depends on `engine_repo_tools`, which already has a way of finding the most recently generated `out/` directory: https://github.com/flutter/engine/blob/main/tools/pkg/engine_repo_tools/lib/engine_repo_tools.dart#L159, so that could be a good way to avoid adding similar logic in the shell script.